app/assets/javascripts/admin/index_utils/directives/obj_for_update.js.coffee

```diff
@@ -11,14 +11,9 @@ angular.module("admin.indexUtils").directive "objForUpdate", (switchClass, pendi
         pendingChanges.remove(scope.object().id, scope.attr)
         scope.clear()
       else
-        change =
-          object: scope.object()
-          type: scope.type
-          attr: scope.attr
-          value: if value? then value else ""
-          scope: scope
         scope.pending()
-        pendingChanges.add(scope.object().id, scope.attr, change)
+        value = if value? then value else ""
+        addPendingChange(scope.attr, value)
```

Comment:
This refactor looks good, though perhaps it could have been a separate commit first.

The only thing is I notice we're updating an argument variable, which is probably not a good idea. We can probably just condense to one line.
I'm not familiar with the coffeescript `?` modifier, but assume it means "truthy", so would this be equivalent?

```suggestion
        addPendingChange(scope.attr, value || "")
```